GooglePlayInstant/Editor/QuickDeploy/DialogHelper.cs

```diff
@@ -64,5 +63,55 @@ public static string SaveFilePanel(string title, string defaultPath, string exte
 
             return EditorUtility.SaveFilePanel(title, directory, fileName, extension);
         }
+
+        /// <summary>
+        /// Displays a save dialog pointing to the default path,
+        /// and requires the user to select a path within the Assets folder.
+        /// If the path points to a file then that file will be used as the default file name.
+        /// </summary>
+        /// <returns>The user selected path relative to the Assets folder, or null if the dialog is closed</returns>
+        public static string SaveFilePanelInProject(string title, string defaultPath, string extension)
+        {
+            // We use this instead of EditorUtility.SaveFilePanelInProject,
+            // because that function doesn't allow us to specify a default path.
+
+            string saveFilePath = null;
+            while (saveFilePath == null)
+            {
+                saveFilePath = SaveFilePanel(title, defaultPath, extension);
+                if (String.IsNullOrEmpty(saveFilePath))
+                {
+                    // Assume cancelled.
+                    return null;
+                }
+
+                saveFilePath = AbsoluteToAssetsRelativePath(saveFilePath);
+                if (String.IsNullOrEmpty(saveFilePath))
+                {
+                    DisplayMessage("Need to save in the Assets folder",
+                        "You need to save the file inside of the project's assets folder");
+                }
+            }
+
+            return saveFilePath;
+        }
+
+        /// <summary>
+        /// Converts the specified absolute path to a path relative to the Assets folder.
+        /// Returns null if the path does not contains the Assets folder.
+        /// </summary>
+        private static string AbsoluteToAssetsRelativePath(string absolutePath)
+        {
+            var parentPath = Application.dataPath + Path.DirectorySeparatorChar;
+            var index = absolutePath.IndexOf(parentPath, StringComparison.Ordinal);
+            if (index != 0)
+            {
+                return null;
+            }
+
+            var relativePath = absolutePath.Remove(index, parentPath.Length);
+            relativePath = Path.Combine("Assets/", relativePath);
+            return relativePath;
```

Comment:
Maybe just return directly from the statement on the previous line?

Reply:
Good call.